src/routes/kyc.ts

```diff
@@ -12,12 +12,15 @@ import {
   kycRequestParamsSchema,
   KycSchema,
   personalDataAndDocumentsKycSchema,
+  personalDataAndDocumentsDetailedKycSchema,
 } from '@fiatconnect/fiatconnect-types'
-import { AnyZodObject } from 'zod'
+import { ZodTypeAny } from 'zod'
 
-const kycSchemaToZodSchema: { [kycSchema in KycSchema]: AnyZodObject } = {
+const kycSchemaToZodSchema: { [kycSchema in KycSchema]: ZodTypeAny } = {
```

Comment:
this was an extremely maddening change to realize we needed. as it turns out, `personalDataAndDocumentsDetailedKycSchema` is actually of type `ZodIntersection`, since the [chain of operations on the base object](https://github.com/fiatconnect/fiatconnect-types/blob/main/src/kyc.ts#L103) we use to construct it ends with an `and`.

this information is buried [extremely deep](https://zod.dev/?id=intersections) in the zod docs, and helpfully _not_ in the section actually describing the `.and` method. as it turns out, certain zod methods require the arguments to be an actual `ZodObject`; some require a raw `ZodRawShape`, and few, if any, support `ZodIntersection` or `ZodUnion` (both of which are relevant to us, given how we construct the schema). none of this is documented. after spending too long trying to wrangle this schema in `fiatconnect-types` into an actual `ZodObject` so the typechecking here would pass, i discovered the `ZodTypeAny` type, which afaik, should be perfectly acceptable to use here.

Reply:
thanks for researching and explaining. might be worth opening an issue for them to fix or at least update their docs